apps/web-portal-client/lib/features/home/hooks/usePreviews.jsx

```diff
@@ -0,0 +1,48 @@
+import {useContext, useReducer} from 'react'
+import {AppContext} from '../../../context'
+
+export const usePreviews = () => {
+  const fetchPreviewsReducer = (state, action) => {
+    switch (action.type) {
+      case 'FETCH_INIT': {
+        return {...state, isLoading: true, isError: false}
+      }
+      case 'FETCH_SUCCESS': {
+        return {...state, isLoading: false, previews: action.payload}
+      }
+      case 'FETCH_FAILURE': {
+        return {...state, isLoading: false, isError: true}
+      }
+      default:
+        return state
+    }
+  }
+
+  // TODO: extract as reusable reducer for HTTP calls
```

Comment:
looks like we I've requested few similar TODOs, plz remove this one, because we would cover common reducer in providing common fetch hook